### PR TITLE
Correct the filtering logic and add a test case

### PR DESCRIPTION
**Pull request by Randy-Ma**
### Purpose

This submission is to correct the filtering logic which is made when I am fixing failing test cases in https://github.com/DynamoDS/DynamoRevit/pull/374.

A test case has been added to track regressions.

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] The level of testing this PR includes is appropriate

### Reviewers

@aparajit-pratap

### CODE REVIEW

File: src/Libraries/RevitNodesUI/Selection.cs [hunk 1]
```diff
@@ -227,7 +227,7 @@ protected override string GetIdentifierFromModelObject(TSelection modelObject)
 
             // If the deleting operations does not make any elements in SelectionResults
             // invalid, then there is no need to update.
-            if (SelectionResults.Select(el => !el.IsValidObject).Count() == 0)
+            if (SelectionResults.Where(el => !el.IsValidObject).Count() == 0)
```
Comment (Randy-Ma) [inline src/Libraries/RevitNodesUI/Selection.cs]: @aparajit-pratap 
Here, Where should be used.

Comment (aparajit-pratap) [inline src/Libraries/RevitNodesUI/Selection.cs]: Oh yes, I overlooked this. Wonder how it passed the tests?

File: test/Libraries/RevitIntegrationTests/BugTests.cs [hunk 2]
```diff
@@ -621,6 +622,85 @@ public void MAGN_6710()
...
+            using (var trans = new Autodesk.Revit.DB.Transaction(document, "DeleteReferencePoint"))
+            {
+                trans.Start();
+                document.Delete(lineID);
+                trans.Commit();
+            }
+
+            RunCurrentModel();
+            //There should be no infinite loop, otherwise, there will be an error with this test case.
+        }
```
Comment (aparajit-pratap) [inline test/Libraries/RevitIntegrationTests/BugTests.cs]: So this test just checks if the deletion of a Revit element happens successfully?

Comment (Randy-Ma) [inline test/Libraries/RevitIntegrationTests/BugTests.cs]: This is to check whether an update in the graph, e.g., deleting an element will cause the graph to  run endlessly.

File: src/Libraries/RevitNodesUI/Selection.cs (hunk 1 above)
Comment (Randy-Ma) [inline src/Libraries/RevitNodesUI/Selection.cs]: Because it somehow avoids the call to get the element ID for invalid objects.

**Comment by aparajit-pratap** [comment]
In addition you could check if there is no change in state of the graph at the end of the test case, i.e. the graph has not re-executed since that is what we need to essentially test here. Apart from this minor comment, LGTM.

**Comment by aparajit-pratap** [comment]
Please give a thought to the declarations made in the PR description and whether they are true.

**Pull request merged**